### ISSUE TITLE
System.NullReferenceException on startup

### ISSUE THREAD

**Issue by hez2010** [AvaloniaUI/Avalonia#3271]
Environment: WSL (ubuntu 18.04):
I've configured Xming for GUI support in WSL.

Steps:
```
dotnet new avalonia.mvvm -o MyApp
cd MyApp
dotnet run
```

Failed to start:
```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at Avalonia.Shared.PlatformSupport.StandardRuntimePlatform.UnmanagedBlob.DoDispose()
   at Avalonia.Shared.PlatformSupport.StandardRuntimePlatform.UnmanagedBlob.Finalize()
```

**Comment by PatrickKursawe** [comment]
Same here.

**Comment by josegomez** [comment]
I'm running into this same exact issue. Any updates?

**Comment by josegomez** [comment]
Doing some research on this with @danwalmsley  Here's what we've found so far
![image](https://user-images.githubusercontent.com/420837/72654748-e7fc4480-395e-11ea-8169-9e9e1d9089d4.png)

Appears that size is reported as 0 (probably bc of the Export Display?)
This loops for ever throwing that exception, eventually it tries to Free _plat with a size 0 and throws another un-recoverable exception
![image](https://user-images.githubusercontent.com/420837/72654782-0a8e5d80-395f-11ea-828d-e9dc45fbe027.png)

**Comment by danwalmsley** [comment]
Seems like it throws an exception trying to allocate an unmanagedblob size 0, and then when it calls DoDispose _plat is null, hence the NRE.

I wonder if there is a lesson about not throwing inside a ctor? Here it ends up leaving a partially constructed object?

**Comment by mausch** [comment]
I think this may have been fixed in https://github.com/AvaloniaUI/Avalonia/pull/3453 ?

**Comment by hez2010** [comment]
> I think this may have been fixed in #3453 ?

Yes. I can confirm that this issue has been fixed.
